hack/test/e2e-docker.sh

```diff
@@ -14,7 +14,8 @@ function create_cluster {
     --provisioner "${PROVISIONER}" \
     --name "${CLUSTER_NAME}" \
     --image "${IMAGE}" \
-    --masters=3 \
+    --masters=1 \
```

Comment:
docker mode doesn't use loadbalancer, and control plane endpoint points to the first (bootstrap node)

without a loadbalancer, there's always a chance that control plane pods are not running on bootstrap node by the time bootkube tears down boostrap pods (but it is running on other control plane nodes). so running HA control plane in docker mode is kind of pointless and leading to flakes